components/right-sidebar/SocialMediaModal.tsx

```diff
@@ -0,0 +1,144 @@
+"use client";
+
+import { Loader2, X } from "lucide-react";
+import {
+  SubmitHandler,
+  useForm,
+  Controller,
+  FieldError,
+} from "react-hook-form";
+import toast from "react-hot-toast";
+import { zodResolver } from "@hookform/resolvers/zod";
+
+import { updateUser } from "@/lib/actions/user.actions";
+import {
+  ISocialMediaSchema,
+  IUserSchema,
+  SocialMediaSchema,
+} from "@/lib/validations/UserSchema";
+import { Social, User } from "@prisma/client";
+import Input from "../shared/ui/Input";
+import { socialMediaIconList } from "@/lib/constants/socialMediaList";
+import Button from "../shared/ui/Button";
+
+const SocialMediaModal = ({
+  user,
+  onClose,
+}: {
+  user: User & { socialMedia?: Social[] };
+  onClose: () => void;
+}) => {
+  const useFormHelpers = useForm<ISocialMediaSchema>({
+    defaultValues: {
+      socialMedia: user.socialMedia ?? [],
+    },
+    resolver: zodResolver(SocialMediaSchema),
+  });
+
+  const {
+    formState: { isSubmitting, errors },
+    control,
+    trigger,
+    handleSubmit,
+    register,
+  } = useFormHelpers;
+  console.log("errors", errors);
+
+  const onSubmit: SubmitHandler<Partial<IUserSchema>> = async (data) => {
+    console.log("data", data);
```

Comment:
```suggestion

  const onSubmit: SubmitHandler<Partial<IUserSchema>> = async (data) => {
```
I would remove all these console logs from testing before we go ahead and try and get this merged into main